### ISSUE TITLE
feat(python): PyO3 bindings for Bragg edge and ray tracing modules

### ISSUE THREAD

**Issue by KedoKudo** [ornlneutronimaging/NEREIDS#457]
Part of the Bragg edge epic (#451).
Depends on: all previous sub-issues.

## What

Expose nereids-bragg and nereids-raytrace through the existing Python bindings
(`bindings/python/src/lib.rs`).

### Cross-section computation

```python
import nereids
import numpy as np

wavelengths = np.linspace(0.5, 6.0, 2000)  # Angstroms

# Powder-average cross-section
sigma = nereids.bragg_cross_section(
    wavelengths,
    lattice_a=2.8665,
    crystal_system='bcc',
    b_coh=9.45,
    temperature_k=300.0,
)
# Returns: 1D numpy array, barns per atom

# Oriented single-crystal cross-section
sigma = nereids.bragg_cross_section(
    wavelengths,
    lattice_a=2.8665,
    crystal_system='bcc',
    b_coh=9.45,
    orientation=[0.707, 0.0, 0.707, 0.0],  # quaternion [w, x, y, z]
    beam_direction=[0.0, 0.0, 1.0],
    strain_tensor=np.zeros((3, 3)),         # optional
    temperature_k=300.0,
)

# Total cross-section (coherent elastic + absorption + incoherent + inelastic)
sigma_total = nereids.bragg_total_cross_section(
    wavelengths,
    lattice_a=2.8665,
    crystal_system='bcc',
    b_coh=9.45,
    sigma_abs=2.56,       # barns at 1.798 A
    sigma_incoh=0.40,     # barns
    debye_temperature=470.0,
    temperature_k=300.0,
)
```

### Transmission with resolution

```python
# Compose with existing resolution broadening
res = nereids.load_resolution('venus_resolution.dat', flight_path_m=25.0)
transmission = nereids.bragg_transmission(
    wavelengths,
    sigma_total,            # barns/atom from bragg_total_cross_section
    areal_density=0.05,     # atoms/barn (= n * d)
    resolution=res,         # optional
)
```

### Ray tracing

```python
# cross_sections: (Nx, Ny, Nz, N_lambda) float64, units cm^-1
# Returns: (det_y, det_x, N_lambda) transmission
transmission_image = nereids.raytrace_spectral(
    grid_dims=[64, 64, 64],
    grid_size=[1e-5, 1e-5, 1e-5],   # meters
    cross_sections=sigma_4d,
    beam_direction=[0.0, 0.0, 1.0],
)

# Single-wavelength projection (for quick checks)
line_integrals = nereids.raytrace_projection(
    grid_dims=[64, 64, 64],
    grid_size=[1e-5, 1e-5, 1e-5],
    values=sigma_3d,                 # (Nx, Ny, Nz) float64, cm^-1
    beam_direction=[0.0, 0.0, 1.0],
)
# Returns: (det_y, det_x) float64
```

## Implementation notes

- All functions release the GIL during computation (use `py.detach()` pattern
  from existing bindings)
- Use `PyReadonlyArray` for input arrays, `PyArray` for output
- Crystal system as string: 'bcc', 'fcc', 'hcp' (with optional c_over_a kwarg)
- Orientation as list/array of 4 floats (auto-normalized to unit quaternion)
- Strain tensor as 3×3 numpy array (validated for symmetry)

## Tests

- Python-level smoke tests mirroring the Rust unit tests
- Round-trip: compute cross-section, apply Beer-Lambert, verify T = exp(−n×σ×d)
- Resolution broadening: verify Bragg edges are smoothed when resolution is applied
- Ray tracing: uniform grid gives expected transmission

**Comment by KedoKudo** [comment]
Closing as out of the frozen Phase-5 scope (contract PR #724, epic #725): the pipeline-map contract fixes NEREIDS's scope to resonance transmission imaging in the resolved region, and its closed-surface rule makes any new capability a contract amendment first. Bragg-edge/ray-tracing simulation is a coherent future direction, but keeping it as an open backlog invites the death-by-features drift the contract exists to prevent. Reopen behind a contract amendment when the refactor is complete.
